pennylane/devices/execution_config.py

```diff
@@ -54,6 +54,15 @@ class ExecutionConfig:
     True indicates a request to either use the device gradient or fail.
     """
 
+    use_device_jacobian_product: Optional[bool] = None
```

Comment:
Note that in discussion with @albi3ro and myself, we decided to call this `device_vjp` :+1:

Reply:
@trbromley Is it ok to make the `qml.execute` keyword argument `device_vjp` but keep the `ExecutionConfig` property named `use_device_jacobian_product`? Then it would mimic the `use_device_gradient` property.